nistats/first_level_model.py

```diff
@@ -557,3 +568,214 @@ def compute_contrast(self, contrast_def, stat_type=None,
             '%s of contrast %s' % (output_type, contrast_name))
 
         return output
+
+
+def first_level_models_from_bids(dataset_path, task_id, model_init=None,
+                                 preproc_variant=None, preproc_space=None):
+    """Create FirstLevelModel objects and fit arguments from a BIDS dataset.
+
+    Parameters
+    ----------
+    dataset_path: str
+        Directory of the highest level folder of the BIDS dataset. Should
+        contain subject folders and a derivatives folder.
+
+    task_id: str
+        Task_id as specified in the file names. Of the form _task-*_.
+
+    model_init: dict, optional
```

Comment:
As long as the BIDS norm does not say anything, I'd be in favor of explicit arguments. It is very likely that BIDS will come with a specific for these arguments, and then we'll use the API they suggest (e.g. a json file).
But I really want to avoid unconstrained dictionaries, because it is impossible to assert whether the argument is acceptable or not.